internals/daemon/api_notices.go

```diff
@@ -267,19 +249,22 @@ func v1GetNotice(c *Command, r *http.Request, _ *UserState) Response {
 	if notice == nil {
 		return NotFound("cannot find notice with ID %q", noticeID)
 	}
-	if !noticeViewableByUser(notice, requestUID, daemonUID) {
+	if !noticeViewableByUser(notice, user) {
 		return Forbidden("not allowed to access notice with id %q", noticeID)
 	}
 	return SyncResponse(notice)
 }
 
-func noticeViewableByUser(notice *state.Notice, requestUID, daemonUID uint32) bool {
+func noticeViewableByUser(notice *state.Notice, user *UserState) bool {
 	userID, isSet := notice.UserID()
 	if !isSet {
+		// Notice has no UID, so it's viewable by any user.
 		return true
```

Comment:
But not readable to user's without a UID.

Reply:
I've tweaded the comment, and added a TODO comment in https://github.com/canonical/pebble/pull/434/commits/bc5d32d6acd44977d39c4f92b5290866707103e4